elfi/methods/methods.py

```diff
@@ -478,18 +476,28 @@ def update(self, batch, batch_index):
         self._update_objective()
 
     def extract_result(self):
-        """Extracts the result from the current state"""
+        """Extracts the result from the current state
+
+        Returns
+        -------
+        result : Result
+        """
         if self.state['samples'] is None:
             raise ValueError('Nothing to extract')
 
         # Take out the correct number of samples
-        samples = dict()
-        for k, v in self.state['samples'].items():
-            samples[k] = v[:self.objective['n_samples']]
+        n_samples = self.objective['n_samples']
+        samples = [self.state['samples'][p][:n_samples]
+                   for p in self.outputs]
```

Comment:
Wondering the change here. This potentially changes the behaviour, where the earlier took all the keys from `samples`, and this only `self.outputs`.